enhancements/rhcos/liveisoinstall.md

```diff
@@ -0,0 +1,95 @@
+---
+title: RHEL CoreOS Live ISO/installer
+authors:
+  - "@cgwalters"
+reviewers:
+  - "@coreos-team"
+approvers:
+  - "@coreos-team"
+creation-date: 2020-02-07
+last-updated: 2020-02-07
+status: implementable
+---
+
+# Rebase RHEL CoreOS 4.5 on Fedora CoreOS $latest
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RHEL CoreOS is built from RHEL components, but uses configuration and tooling from Fedora CoreOS.  They share many things such as the installer code, etc.
+
+This enhancement tracks rebasing RHCOS 4.5 on the latest FCOS code, in particular the installer.
+
+### New "Live" installer
+
+Currently, RHCOS 4.4 and below offer a kernel/initramfs and ISO that can run a shell script installation program from the initramfs.
+
+In FCOS $latest, there is a Live ISO/PXE media that runs the full OS (for example, it includes `podman`), and a [coreos-installer](https://github.com/coreos/coreos-installer/) program was rewritten.
+
+## Motivation
+
+This allows administrators to run arbitrary code before and after the installation process by providing Ignition that wraps the installer.  For example, higher level tooling like Foreman and Ironic can easily `curl` a URL to report provisioning success, or even pull down and launch a container before/after.
```

Comment:
As a product (OpenShift) don't we want to discourage this (modifying ignition). As a product, we may use an underlying technology (from upstream) to accomplish a goal, but not neccarly expose that to the downstream products end-users. (this is how I view ignition - cool upstream tech, that we treat as a black box, inside of the installer)

If we allow this (arbitrary modification of ignition) how will we keep 'ignition' from becoming the next 'ansible' (issue for OpenShift); that makes the infinite number of configuration permutations, impossible for us to comprehend / track; and thus can't ensure a smooth/simple install experience? Is there a way to gate/validate the ignition we get to make sure its safe and does not conflict with our install processes, befor the installer starts?

Reply:
We have to support appending to the openshift-installer generated Ignition today for *machine-specific* configuration like `/etc/hostname` and networking etc.

> If we allow this (arbitrary modification of ignition) how will we keep 'ignition' from becoming the next 'ansible' (issue for OpenShift); that makes the infinite number of configuration permutations

Today MachineConfig via the MCO *already* allows executing arbitrary code as root via systemd units - day 1 or day 2.  And it's the same thing with Ignition.

> Is there a way to gate/validate the ignition we get to make sure its safe and does not conflict with our install processes, befor the installer starts?

I can't really think of a way to do that other than to provide a higher level set of knobs, you could imagine something a bit like https://github.com/coreos/fcct for provisioning time, something like a file:

```
postInstallURL: http://example.com/provisioning/blah
```

that openshift-install could perhaps accept and generate the Ignition for the machine install.

Reply:
Note that the Ignition config described in this paragraph is not the config applied to the installed system.  It's a second, independent Ignition config that only applies to the live system which runs the installer itself.